meilisearch/tests/search/errors.rs

```diff
@@ -37,104 +38,360 @@ async fn search_unexisting_parameter() {
+      "code": "invalid_search_limit",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-limit"
+    }
+    "###);
+
+    let (response, code) = index.search_get(json!({"limit": "doggo"})).await;
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "invalid digit found in string at `.limit`.",
+      "code": "invalid_search_limit",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-limit"
+    }
+    "###);
+}
+
+#[actix_rt::test]
+async fn search_bad_page() {
+    let server = Server::new().await;
+    let index = server.index("test");
+
+    let (response, code) = index.search_post(json!({"page": "doggo"})).await;
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "invalid type: String `\"doggo\"`, expected a Integer at `.page`.",
+      "code": "invalid_search_page",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-page"
+    }
+    "###);
+
+    let (response, code) = index.search_get(json!({"page": "doggo"})).await;
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "invalid digit found in string at `.page`.",
+      "code": "invalid_search_page",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-page"
+    }
+    "###);
+}
+
+#[actix_rt::test]
+async fn search_bad_hits_per_page() {
+    let server = Server::new().await;
+    let index = server.index("test");
+
+    let (response, code) = index.search_post(json!({"hitsPerPage": "doggo"})).await;
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "invalid type: String `\"doggo\"`, expected a Integer at `.hitsPerPage`.",
+      "code": "invalid_search_hits_per_page",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-hits-per-page"
+    }
+    "###);
+
+    let (response, code) = index.search_get(json!({"hitsPerPage": "doggo"})).await;
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "invalid digit found in string at `.hitsPerPage`.",
+      "code": "invalid_search_hits_per_page",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-hits-per-page"
+    }
+    "###);
+}
+
+#[actix_rt::test]
+async fn search_bad_attributes_to_crop() {
+    let server = Server::new().await;
+    let index = server.index("test");
+
+    let (response, code) = index.search_post(json!({"attributesToCrop": "doggo"})).await;
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "invalid type: String `\"doggo\"`, expected a Sequence at `.attributesToCrop`.",
```

Comment:
Other future improvement: expected a Sequence of what?
Maybe we could customise these messages with something like:
```rust
struct Search {
    #[deserr(expected = "an array containing the names (String) of the attributes to crop")]
    attributes_to_crop: Vec<String>
}
```

Reply:
ah yeah that would be nice